gorm/transaction.go

```diff
@@ -20,6 +21,12 @@ type ctxKey int
 // instead of using this key directly.
 var txnKey ctxKey
 
+type CtxTxnMissingError struct{}
```

Comment:
Did you add new type to provide a way for developer to distinguish transaction error?
If so, I would change it on public variable instead.